### ISSUE TITLE
Cleverspa Hot Tub?

### ISSUE THREAD

**Issue by frizby66** [0lly18/CleverSpa_Experimental#2]
Hi, were you trying to create a homebridge plugin to link a wifi enabled Cleverspa to homebridge?

**Comment by 0lly18** [comment]
Hi, I started to look at it but didn't have the time to learn enough to make it work. I ended up using the [Homebridge Http Switch](https://github.com/Supereg/homebridge-http-switch#readme) plugin (to control the heater, filter and bubbles) and [Homebridge Htttp Temperature Sensor](https://github.com/Supereg/homebridge-http-temperature-sensor#readme) to get the current temperature.

**Comment by frizby66** [comment]
Interesting…assume you still have your tub and was it a native wifi enabled tub or did you use a different device to control?

I am one of the admins on Cleverspa Users FB Group (no connection with Cleverspa) and one of the members (Dylan) is doing some amazing stuff with his own website, which can do lots of cool things, including scheduling and data logging, etc. all using the api.

How did you configure the plugins, did you use “as is” or did you modify? Care to share the entire for the config.json (removing tokens, did, etc.)?

Fraser

**Comment by 0lly18** [comment]
yeah still use the tub, it's useful to have the temp and switches in HomeKit rather than having to use the Cleverspa app.

That sounds interesting, will have to give the FB Group a look.

I just used the plugins just call the API the app uses, took a bit of trial and error to get them configured but they've been pretty solid so far, here's the config for one of the switches and the temp sensor; 

{
            "accessory": "HTTP-TEMPERATURE",
            "name": "Hot Tub Temperature",
            "unit": "celsius",
            "getUrl": {
                "url": "http://api.gizwits.com/app/devdata/xxxx/latest",
                "method": "GET",
                "headers": {
                    "Content-Type": "application/json",
                    "X-Gizwits-User-token": "xxxx",
                    "X-Gizwits-Application-Id": "xxxx"
                }
            },
            "statusPattern": "\"Current_temperature\":([0-9]{2})",
            "patternGroupToExtract": 1,
            "debug": "false"
        },
        {
            "accessory": "HTTP-SWITCH",
            "name": "Hot Tub Heater",
            "switchType": "stateful",
            "debug": "false",
            "onUrl": {
                "url": "http://api.gizwits.com/app/control/xxxx",
                "method": "POST",
                "body": {
                    "attrs": {
                        "Heater": 1
                    }
                },
                "headers": {
                    "Content-Type": "application/json; charset=utf-8",
                    "X-Gizwits-User-token": "xxxx",
                    "X-Gizwits-Application-Id": "xxxx"
                }
            },

The other switches, Filter and Bubbles, are just a copy and paste of the heater config just replacing "Heater" with the appropriate attribute

**Comment by frizby66** [comment]
Excellent…thanks for that…trying to get a mate of mine to pull the temp sensing, temp set and heater on/off into the one plugin (it would mimic a thermostat in Homekit) to see if we can get it to work. I don't have a need for filter/bubbles via Homekit.

Have attached a screen shot off what Dylan is doing…I am beta testing the site for him. You will see a ~ next to the the second schedule time…this means that you want that target temperature set at that time and the site monitors the current temp v warm up time (degrees per hour) and calculates when it needs to raise the temp on the tub to match the desired temp by the desired time.

So everyday, I set the temp to 20 degrees at night and then on a Friday, Saturday and Sunday, I set the tub to be ready for 11:30 in the morning.

**Comment by frizby66** [comment]
Quick question, when I install the plugin and update the config file, I get a json error on the line

"statusPattern": ""Current_temperature":([0-9]{2})”,

It says it is not the correct syntax. Did you get the same?

**Comment by frizby66** [comment]
Found this issue…

This worked

 "statusPattern": ".Current_temperature.:([0-9]{2})",

**Comment by 0lly18** [comment]
yeah, this drove me a bit mad trying to figure our the correct regex
